src/store/selectors/members.ts

```diff
@@ -24,3 +28,81 @@ export function getMembersByUsernames(
     state.members.byMemberUsername.get(username, undefined)
   );
 }
+
+function getAncestors(
+  member: Member,
+  byMemberId: ImmutableMap<MemberId, Member>
+): Set<MemberId> {
+  // Everyone is their own ancestor in this function
+  let votingForMember = member as Member | undefined;
+  const ancestors = new Set<MemberId>();
+  for (
+    let votingForId: MemberId | typeof GENESIS_MEMBER = member.memberId;
+    votingForId !== GENESIS_MEMBER; // TODO needs to change to check if voting for self
+    votingForId = votingForMember.invitedBy // TODO needs to change to votingFor
+  ) {
+    votingForMember = byMemberId.get(votingForId);
+    if (votingForMember === undefined) {
+      throw Error(`Cannot vote for invalid uid ${votingForId}`);
+    }
+    if (ancestors.has(votingForId)) {
+      throw Error(
+        `Cycle in ancestors of ${member.memberId}: ${votingForId} found twice`
+      );
+    }
+    ancestors.add(votingForId);
+  }
+  return ancestors;
+}
+
+function incrementKey(counts: Map<any, number>, key: any) {
+  const count = counts.get(key);
+  counts.set(key, count === undefined ? 1 : count + 1);
+}
+
+function incVotesForAncestors(
+  member: Member,
+  membersByUid: ImmutableMap<MemberId, Member>,
+  votesByUid: Map<MemberId, number>
+) {
+  const ancestors = getAncestors(member, membersByUid);
+  ancestors.forEach(a => incrementKey(votesByUid, a));
+}
+
+function getSortedMembersAndScore(state: RahaState, rankFn: (member: Member) => number): [Member, number][] {
```

Comment:
has this file been auto-formatted? I feel like this line would be normally wrapped.